pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -703,39 +687,41 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *v1.Serv
 
 		glog.V(4).Infof("Pool for listener %s: %s", listener.ID, pool.ID)
 		members, err := getMembersByPoolID(lbaas.network, pool.ID)
-		if err != nil && !isNotFound(err) {
+		if err != nil && err != ErrNotFound {
 			return nil, fmt.Errorf("Error getting pool members %s: %v", pool.ID, err)
 		}
-		for _, node := range nodes {
-			addr, err := nodeAddressForLB(node)
-			if err != nil {
-				if err == ErrNotFound {
-					// Node failure, do not create member
-					glog.Warningf("Failed to create LB pool member for node %s: %v", node.Name, err)
-					continue
-				} else {
-					return nil, fmt.Errorf("Error getting address for node %s: %v", node.Name, err)
+		if !isNotFound(err) && members != nil {
```

Comment:
You have changed this code to separately use both a literal `err == ErrNotFound` *and* `isNotFound(err)`.  You should pick one :)

I suggest leaving this to use `isNotFound()` and remove all the places above where you've pushed this logic into the `getFooByBar` functions - because it's going to be a lot more fragile to rely on everyone returning just the `ErrNotFound` value.